BUILDING.md

```diff
@@ -53,66 +29,169 @@ following packages are installed:
 
 ## Library dependencies
 
+### Build Dependencies
+
 Binary builds additionally require the following packages to be
-available to the CMake build system.
+available.
 
 - Python 3.7 (development install)
 - [pybind11](https://pybind11.readthedocs.io/en/stable/) 2.6.1+
```

Comment:
[consider] Not sure that pybind11 will still work, we're currently using 2.8.1 (but then that'll change too once we have MacOS 11 support from #681)

Reply:
Will update version to 2.8.1 for now.

Reply:
[0e3ee0b](https://github.com/OpenAssetIO/OpenAssetIO/pull/711/commits/0e3ee0bba5ca3470c7a46e8427f23ffc951e49dc)

Reply:
> [consider] Not sure that pybind11 will still work, we're currently using 2.8.1 (but then that'll change too once we have MacOS 11 support from #681)

*Python 3.11